src/orders/utils/data.ts

```diff
@@ -62,17 +61,17 @@ export function getWarehousesFromOrderLines<T extends OrderLineWithStockWarehous
             ? warehouses
             : [...warehouses, stock.warehouse],
         warehouses,
-      ),
+      ) ?? warehouses,
     [] as WarehouseFragment[],
   );
 }
 
 export function getPreviouslyRefundedPrice(order: OrderRefundSharedType): IMoney {
-  return (
-    getOrderCharged(order as OrderDetailsFragment) &&
-    order?.total?.gross &&
-    subtractMoney(getOrderCharged(order as OrderDetailsFragment), order?.total?.gross)
-  );
+  const orderDetails = order as OrderDetailsFragment;
+  const charged = getOrderCharged(orderDetails);
+  const total = orderDetails?.total?.gross;
+
+  return charged && total ? subtractMoney(charged, total) : charged;
```

Comment:
can you please add test? this is where we should be aligned - we don't need to test html but the fact there was a bug, we should ensure its tested now